internal/flowrate/monitor.go

```diff
@@ -0,0 +1,64 @@
+package flowrate
+
+import (
+	"time"
+
+	legacyimpl "github.com/cometbft/cometbft/internal/flowrate/legacy"
+	"github.com/cometbft/cometbft/internal/flowrate/xrate"
+)
+
+var DefaultImplementation = "xrate"
+
+type Status = legacyimpl.Status
```

Comment:
Actually, in order for the dependencies to be cleaner, I'd move the Status definition (which is common to both implementations) here, and then would import it  in `legacy`

Reply:
You would create a loop :)